mcs/class/System.Private.CoreLib/System/Type.cs

```diff
@@ -104,4 +117,386 @@ internal virtual string FormatTypeName (bool serialization)
 			return !object.ReferenceEquals (left, right);
 		}
 	}
+
+	internal sealed class TypeNameParser
```

Comment:
Make it static and move to separate file

Reply:
Fixed.